### PR TITLE
PLT-2905 fixing upgrade of SSO accounts

### PR DESCRIPTION
**Pull request by coreyhulen**
This also has a corresponding PR in the enterprise repo.

### CODE REVIEW

File: mattermost.go [hunk 1]
```diff
@@ -530,13 +545,19 @@ func convertTeamTo30(primaryTeamName string, team *TeamForUpgrade, uniqueEmails
...
+
 		if shouldUpdateUser {
 			if _, err := store.GetMaster().Exec(`
 				UPDATE Users 
 				SET 
 				    Email = :Email,
 				    Username = :Username,
-				    Roles = :Roles
+				    Roles = :Roles,
+				    AuthService = '',
```
Comment (enahum) [inline mattermost.go]: I don't have this very clear, should the update User set AuthService to what he had and AuthData to match the new email address?

Comment (coreyhulen) [inline mattermost.go]: This is the case where we are changing the email or username and breaking the connection to LDAP or Gitlab.  In that case we blank the AuthData and AuthService fields to force it back to email login for duplicate accounts.

**Comment by jwilander** [comment]
+1

**Comment by enahum** [comment]
+1

**Comment by crspeller** [comment]
+1

**Pull request merged**